crates/cast/src/lib.rs

```diff
@@ -1364,6 +1364,58 @@ impl SimpleCast {
         Ok(formatted)
     }
 
+    /// Convert a number into a uint with arbitrary decimals.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// # fn main() -> eyre::Result<()> {
+    /// assert_eq!(Cast::parse_units("1.0", 6)?, "1000000"); // USDC (6 decimals)
+    /// assert_eq!(Cast::parse_units("2.5", 6)?, "2500000");
+    /// assert_eq!(Cast::parse_units("1.0", 12)?, "1000000000000"); // 12 decimals
+    /// assert_eq!(Cast::parse_units("1.23", 3)?, "1230"); // 3 decimals
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn parse_units(value: &str, unit: u8) -> Result<String> {
+        let unit = Unit::new(unit).ok_or_else(|| eyre::eyre!("invalid unit"))?;
+
+        Ok(ParseUnits::parse_units(value, unit)?.to_string())
+    }
+
+    /// Format a number from smallest unit to decimal with arbitrary decimals.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// # fn main() -> eyre::Result<()> {
+    /// assert_eq!(Cast::format_units("1000000", 6)?, "1"); // USDC (6 decimals)
+    /// assert_eq!(Cast::format_units("2500000", 6)?, "2.500000");
+    /// assert_eq!(Cast::format_units("1000000000000", 12)?, "1"); // 12 decimals
+    /// assert_eq!(Cast::format_units("1230", 3)?, "1.230"); // 3 decimals
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn format_units(value: &str, unit: u8) -> Result<String> {
+        let value = NumberWithBase::parse_int(value, None)?.number();
+        let unit = Unit::new(unit).ok_or_else(|| eyre::eyre!("invalid unit"))?;
+        let mut formatted = ParseUnits::U256(value).format_units(unit);
```

Comment:
since this format code is the same as

https://github.com/foundry-rs/foundry/blob/dcebcd0336c33eb5917ec38b07d0420f3a3c6071/crates/cast/src/lib.rs#L1354-L1362

could you please add a helper fn
```rust
fn format_unit_as_string(value: U256, unit: Unit) -> String {
    let mut formatted = ParseUnits::U256(value).format_units(unit);
    // Trim empty fractional part.
    if let Some(dot) = formatted.find('.') {
        let fractional = &formatted[dot + 1..];
        if fractional.chars().all(|c: char| c == '0') {
            formatted = formatted[..dot].to_string();
        }
    }
    formatted
}
```

and reuse `Ok(format_unit_as_string(value, unit))` in both `to_unit` and `format_units`